lib/puppet/pops/serialization/to_data_converter.rb

```diff
@@ -188,17 +188,21 @@ def unknown_to_data(value)
     end
 
     def unknown_key_to_string_with_warning(value)
-      str = value.to_s
+      str = unknown_to_string(value)
       serialization_issue(Issues::SERIALIZATION_UNKNOWN_KEY_CONVERTED_TO_STRING, :path => path_to_s, :klass => value.class, :value => str)
       str
     end
 
     def unknown_to_string_with_warning(value)
-      str = value.to_s
+      str = unknown_to_string(value)
       serialization_issue(Issues::SERIALIZATION_UNKNOWN_CONVERTED_TO_STRING, :path => path_to_s, :klass => value.class, :value => str)
       str
     end
 
+    def unknown_to_string(value)
+      value.is_a?(Regexp) ? value.inspect : value.to_s
```

Comment:
Please use `PRegexpType.regexp_to_s` or  `PRegexpType.regexp_to_s_with_delimiters` whenever converting regexp to string. The output from the `inspect` method differ between ruby versions (has caused us a some grief until I cleaned it up everywhere).

Another option would be to use the StringFormatter with defaults (for all types of values, see line 249).